packages/core/src/messagingProtocols/proteus/Utility/GenericMessageMapper.ts

```diff
@@ -0,0 +1,275 @@
+/*
+ * Wire
+ * Copyright (C) 2019 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import {ConversationOtrMessageAddEvent} from '@wireapp/api-client/lib/event/';
+import logdown from 'logdown';
+
+import {
+  GenericMessageType,
+  PayloadBundle,
+  PayloadBundleSource,
+  PayloadBundleState,
+  PayloadBundleType,
+} from '../../../conversation';
+import {
+  AssetContent,
+  ClearedContent,
+  ConfirmationContent,
+  DeletedContent,
+  EditedTextContent,
+  HiddenContent,
+  KnockContent,
+  LocationContent,
+  ReactionContent,
+  TextContent,
+} from '../../../conversation/content';
+
+export class GenericMessageMapper {
+  private static readonly logger = logdown('@wireapp/core/GenericMessageMapper', {
+    logger: console,
+    markdown: false,
+  });
+
+  // TODO: Turn "any" into a specific type (or collection of types) and make the return type more specific based on the
+  // "genericMessage" input parameter.
+  public static mapGenericMessage(
```

Comment:
I'm actually thinking of removing any trace of mapping on the core. 
IMO the core should only transmit raw events to the consumer and the consumer decides what it want to do with it. 

I'll prepare a breaking change PR, that removes all those mappings, but in the meantime please revert this change to avoid conflicts in the future (and actually this function doesn't seem to be used in your PR)

Reply:
Its used in otrMessageAdd.ts under messagingProtocols/proteus/EventHandler

Reply:
Its gone